lib/galaxy/webapps/galaxy/api/groups.py

```diff
@@ -2,61 +2,79 @@
 API operations on Group objects.
 """
 import logging
-from typing import (
-    Any,
-    Dict,
+
+from fastapi import (
+    Body,
+    Path,
 )
 
 from galaxy.managers.context import ProvidesAppContext
 from galaxy.managers.groups import GroupsManager
-from galaxy.schema.fields import EncodedDatabaseIdField
-from galaxy.web import (
-    expose_api,
-    require_admin,
+from galaxy.schema.fields import DecodedDatabaseIdField
+from galaxy.schema.groups import (
+    GroupCreatePayload,
+    GroupIndexListResponse,
+    GroupShowResponse,
 )
 from galaxy.webapps.galaxy.api import (
-    BaseGalaxyAPIController,
     depends,
+    DependsOnTrans,
+    Router,
 )
 
 log = logging.getLogger(__name__)
 
+router = Router(tags=["groups"])
+
 
-class GroupAPIController(BaseGalaxyAPIController):
-    manager = depends(GroupsManager)
+@router.cbv
+class FastAPIGroups:
+    manager: GroupsManager = depends(GroupsManager)
 
-    @expose_api
-    @require_admin
-    def index(self, trans: ProvidesAppContext, **kwd):
-        """
-        GET /api/groups
-        Displays a collection (list) of groups.
-        """
+    @router.get(
+        "/api/groups",
+        summary="Displays a collection (list) of groups.",
+        require_admin=True,
+    )
+    def index(
+        self,
+        trans: ProvidesAppContext = DependsOnTrans,
+    ) -> GroupIndexListResponse:
         return self.manager.index(trans)
 
-    @expose_api
-    @require_admin
-    def create(self, trans: ProvidesAppContext, payload: Dict[str, Any], **kwd):
-        """
-        POST /api/groups
-        Creates a new group.
-        """
+    @router.post(
+        "/api/groups",
+        summary="Creates a new group.",
+        require_admin=True,
+    )
+    def create(
+        self,
+        trans: ProvidesAppContext = DependsOnTrans,
+        payload: GroupCreatePayload = Body(...),
+    ) -> GroupIndexListResponse:
```

Comment:
I was referring to this, the create endpoint returns a single Group response.

```suggestion
    ) -> GroupShowResponse:
```

Reply:
I used `GroupIndexListResponse` for `create` endpoint because it returns a list with a single group response. but I can make it to not create a list but just a group response.

Reply:
I was not expecting a list here... I wonder if it's a bug I introduced or if it was like that in the original API so we need to keep it for backward compatibility I can't remember it... :sweat_smile:

Reply:
It was a list from the original API:
https://github.com/galaxyproject/galaxy/blob/e8a288b2d389b756c651cbe1a5682239dd6233fe/lib/galaxy/managers/groups.py#L77

Reply:
Yes, I checked back until [the original API endpoint was created in 2012](https://github.com/galaxyproject/galaxy/commit/9bcbabfaa9d1a8a70d4e6e7155015a3a4c8d8c97#diff-f36fa9ebf45ae5ff33bbf8f2749b36b9677ee1fb1600026e61d1c68eabf3059dR70) and it also returns a list with one single group... weird... but I guess it needs to be backward compatible, so all good :+1: 

Sorry for the noise!